spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryParsingSyntaxError.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jpa.repository.query;
+
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+
+/**
+ * An exception thrown if the JPQL query is invalid.
+ *
+ * @author Greg Turnquist
+ * @since 3.1
+ */
+class QueryParsingSyntaxError extends ParseCancellationException {
+
+	public QueryParsingSyntaxError() {}
```

Comment:
Can we attach the original SQL query to the exception for more context? Also, this exception should be public so that users can handle this exception.

Reply:
We can definitely attach the original query to this exception.